### PR TITLE
refactor(experimental): graphql: init query processor

### PR DESCRIPTION
**Pull request by buffalojoec**
This change introduces a query processor which evaluates the query provided to
the `RpcGraphQL`'s `query(..)` method and builds a queue of instructions
describing how and when to hit the RPC.

The `visit(..)` function from `graphql` performs a depth-first traversal on the
query's Abstract Syntax Tree (AST). During this traversal, we're keying on
certain fields or inline fragments to decide which RPC call to make and the
proper parameters to provide.

The "call chain" described in the comments in `src/context/processor/index.ts`
is a specialized queue designed to store a set of RPC call instructions at each
entry.

- The first entry in the queue will be all RPC calls required to evaluate all
  root queries - such as an `account(..)` query.
- The subsequent entries are similar instructions, however their key values -
  such as `address` for accounts - point to the parent entry in the queue.

In other words, it's a chain of RPC call instructions that can be evaluated one
at a time, where the result of any chain "link" (instruction set) can be used to
fetch the next one.

The tests in `src/context/__tests__/` provide layouts to demonstrate what these
call chains look like.

The intention is to use the call chain to make all necessary RPC calls ahead of
time to front-run the cache, then use the resolvers to simply load data from the
context's already-hydrated cache.

The `processQuery(..)` function is currently a no-op, but in the next commit of
the stack, I'll activate it by implementing the dequeuing as well as light
refactoring of the accounts schema.

### CODE REVIEW

**Comment by buffalojoec** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2048?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #2045** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2045?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #2046** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2046?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #2047** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2047?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #2048** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2048?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2048?utm_source=stack-comment).

**Comment by steveluscher** [comment]
Giving this a good think. I'll have some thoughts later today.

**Comment by buffalojoec** [comment]
> Giving this a good think. I'll have some thoughts later today.

Thanks. My main concern was it's going to get _real_ verbose and we're kind of doing what `graphql` is already doing, but I don't know any other way to build out the RPC fetches and decide on optimizations such as `getMultipleAccountInfos`.

A couple things to mention as I'm still hacking on this locally:
- I can use `getMAI` and then just `insert` into the cache directly to hydrate account loads.
- This is going to suck when I add support for fragments that aren't in-line.
- This is going to suck to maintain, period.

**Comment by steveluscher** [comment]
> …I don't know any other way to build out the RPC fetches and decide on optimizations such as `getMultipleAccountInfos`.

I can help here. Let me drum up a quick PR and you can take it over.

**Comment by steveluscher** [comment]
OK, example VERY VERY BAD DO PLEASE START OVER FROM SCRATCH implementation in #2063, @buffalojoec.
